controllers/cluster.weave.works/existinginframachine_controller.go

```diff
@@ -781,56 +781,60 @@ func (a *ExistingInfraMachineReconciler) uncordon(ctx context.Context, node *cor
 	return nil
 }
 
-func (a *ExistingInfraMachineReconciler) setNodeAnnotation(ctx context.Context, node *corev1.Node, key, value string) error {
-	err := a.modifyNode(ctx, node, func(node *corev1.Node) {
+func (a *ExistingInfraMachineReconciler) setNodeAnnotation(ctx context.Context, nodeName string, key, value string) error {
+	err := a.modifyNode(ctx, nodeName, func(node *corev1.Node) {
 		node.Annotations[key] = value
 	})
 	if err != nil {
-		return gerrors.Wrapf(err, "Failed to set node annotation: %s for node: %s", key, node.Name)
+		return gerrors.Wrapf(err, "Failed to set node annotation: %s for node: %s", key, nodeName)
 	}
 	return nil
 }
 
-func (a *ExistingInfraMachineReconciler) setNodeProviderIDIfNecessary(ctx context.Context, node *corev1.Node) error {
-	err := a.modifyNode(ctx, node, func(node *corev1.Node) {
-		node.Spec.ProviderID = "wks://" + node.Name
+func generateProviderID(nodeName string) string {
+	return "wks://" + nodeName
+}
+
+func (a *ExistingInfraMachineReconciler) setNodeProviderIDIfNecessary(ctx context.Context, nodeName string) error {
```

Comment:
Nit: the name of this function seems odd as I don't see the `IfNecessary` check in the code.

Reply:
I added back the check.